### PR TITLE
feat: Env support

### PR DESCRIPTION
**Pull request by AlexsJones**
```
apiVersion: core.openfeature.dev/v1alpha1
kind: FeatureFlagConfiguration
metadata:
  name: featureflagconfiguration-sample
spec:
  flagDSpec:
    envs:
    - name: FOO
      value: BAR
  featureFlagSpec: |
    {}
```

image

### CODE REVIEW

File: config/manager/kustomization.yaml [hunk 1]
```diff
@@ -12,5 +12,5 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 images:
 - name: controller
-  newName: controller
-  newTag: latest
+  newName: cnskunkworks/flagd
```
Comment (thschue) [inline config/manager/kustomization.yaml]: did you change this intentionally?

**Review by thschue — APPROVED** [review]
lgtm

File: webhooks/pod_webhook.go [hunk 2]
```diff
@@ -189,6 +184,7 @@ func (m *PodMutator) injectSidecar(pod *corev1.Pod, configMap string, featureFla
 				MountPath: "/etc/flagd",
 			},
 		},
+		Env: featureFlag.Spec.FlagDSpec.Envs,
```
Comment (skyerus) [inline webhooks/pod_webhook.go:187]: We need a nil check prior to this:
```
1.6612509519226854e+09	DEBUG	mutating-pod-webhook	Creating sidecar for pod open-feature-operator-system/
2022/08/23 10:35:51 http: panic serving 172.17.0.1:16896: runtime error: invalid memory address or nil pointer dereference
goroutine 426 [running]:
net/http.(*conn).serve.func1(0x40005c4000)
	/usr/local/go/src/net/http/server.go:1802 +0xe4
panic({0x10be2a0, 0x1fed970})
	/usr/local/go/src/runtime/panic.go:1052 +0x2b4
github.com/open-feature/open-feature-operator/webhooks.(*PodMutator).injectSidecar(0x400069e510, 0x40004d9800, {0x400043e9f0, 0xa}, 0x4000656c90)
	/workspace/webhooks/pod_webhook.go:187 +0x6ac
github.com/open-feature/open-feature-operator/webhooks.(*PodMutator).Handle(0x400069e510, {0x1486cb0, 0x4000466240}, {{{0x40000422a0, 0x24}, {{0x0, 0x0}, {0x4000410168, 0x2}, {0x400041016a, ...}}, ...}})
	/workspace/webhooks/pod_webhook.go:91 +0x7d8
sigs.k8s.io/controller-runtime/pkg/webhook/admission.(*Webhook).Handle(0x40006b4000, {0x1486cb0, 0x4000466240}, {{{0x40000422a0, 0x24}, {{0x0, 0x0}, {0x4000410168, 0x2}, {0x400041016a, ...}}, ...}})
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/webhook/admission/webhook.go:146 +0x84
sigs.k8s.io/controller-runtime/pkg/webhook/admission.(*Webhook).ServeHTTP(0x40006b4000, {0xffff6bfa76f8, 0x4000110050}, 0x4000514600)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/webhook/admission/http.go:99 +0xdb0
github.com/prometheus/client_golang/prometheus/promhttp.InstrumentHandlerInFlight.func1({0xffff6bfa76f8, 0x4000110050}, 0x4000514600)
	/go/pkg/mod/github.com/prometheus/client_golang@v1.11.0/prometheus/promhttp/instrument_server.go:40 +0x9c
net/http.HandlerFunc.ServeHTTP(0x400069e780, {0xffff6bfa76f8, 0x4000110050}, 0x4000514600)
	/usr/local/go/src/net/http/server.go:2047 +0x40
github.com/prometheus/client_golang/prometheus/promhttp.InstrumentHandlerCounter.func1({0x1482be0, 0x40002242a0}, 0x4000514600)
	/go/pkg/mod/github.com/prometheus/client_golang@v1.11.0/prometheus/promhttp/instrument_server.go:101 +0xb4
net/http.HandlerFunc.ServeHTTP(0x400069e960, {0x1482be0, 0x40002242a0}, 0x4000514600)
	/usr/local/go/src/net/http/server.go:2047 +0x40
github.com/prometheus/client_golang/prometheus/promhttp.InstrumentHandlerDuration.func2({0x1482be0, 0x40002242a0}, 0x4000514600)
	/go/pkg/mod/github.com/prometheus/client_golang@v1.11.0/prometheus/promhttp/instrument_server.go:76 +0x94
net/http.HandlerFunc.ServeHTTP(0x400069eae0, {0x1482be0, 0x40002242a0}, 0x4000514600)
	/usr/local/go/src/net/http/server.go:2047 +0x40
net/http.(*ServeMux).ServeHTTP(0x40006b4040, {0x1482be0, 0x40002242a0}, 0x4000514600)
	/usr/local/go/src/net/http/server.go:2425 +0x18c
net/http.serverHandler.ServeHTTP({0x4000224380}, {0x1482be0, 0x40002242a0}, 0x4000514600)
	/usr/local/go/src/net/http/server.go:2879 +0x45c
net/http.(*conn).serve(0x40005c4000, {0x1486d58, 0x40001b27b0})
	/usr/local/go/src/net/http/server.go:1930 +0xb54
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:3034 +0x4ac
```

Comment (AlexsJones) [reply webhooks/pod_webhook.go:187]: thanks great spot, I must have missed this when converting to coreenv https://github.com/open-feature/open-feature-operator/pull/75

File: pkg/utils/flagd-definitions.json [hunk 3]
```diff
@@ -0,0 +1,168 @@
+{
```
Comment (toddbaert) [inline pkg/utils/flagd-definitions.json:1]: Why was this file added?
